.lib/bacon_harvester.js

```diff
@@ -35,9 +35,16 @@ exports.generatePkb = function (platformName, callback) {
         }
         var package_id = '';
         if (list[i].element.provider != data.baconprovider) {
+
           return checklist();
         }
-
+        if (data.type) {
+          let reg = / /;
+          reg.compile(data.type);
```

Comment:
The `compile()` method has been deprecated a very long time ago, why don't you just use `new RegExp(data.type)` ?